File: app/src/app/productStock/productStockService.js
/**
 * Created by Asus on 19/7/2559.
 */
(function() {
  'use strict'
  angular
    .module('app')
    .factory('productStockService',productStockService);

  /** @ngInject */
  function productStockService($resource){
    return $resource('/productStock/:id', { id: '@_id' }, {
      update: {
        method: 'PUT' // this method issues a PUT request
      }});

  }
})();

File: src/main/java/com/smartfarmh2/productStock/ProductStockService.java
package com.smartfarmh2.productStock;

import java.util.List;

/**
 * Created by Asus on 20/7/2559.
 */
public interface ProductStockService {
    ProductStock create (ProductStock productStock);
    ProductStock update (ProductStock productStock);
    void delete (Long id);
    ProductStock getProductStock (Long id);
    List<ProductStock> list();
}
